packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/quanticColoredResultBadge.js

```diff
@@ -46,17 +47,57 @@ export default class QuanticColoredResultBadge extends LightningElement {
   /** @type{boolean} */
   invalidColor = false;
 
+  /** @type {string} */
+  error;
+  validated = false;
+
+  connectedCallback() {
+    getBueno(this).then(() => {
+      if (
+        (!this.label || !Bueno.isString(this.label)) &&
+        (!this.result || !this.field || !Bueno.isString(this.field))
+      ) {
+        console.error(
+          `The ${this.template.host.localName} requires a label or result and a field to be specified.`
+        );
+        this.setError();
+      }
+      if (this.color && !validHEXColor(this.color)) {
+        console.error(`The "${this.color}" color is not a valid HEX color.`);
+        this.setError();
+      }
+      if (!this.color) {
+        console.warn(
+          'The color property has not been specified, the default colors will be used.'
+        );
+      }
+      this.validated = true;
+    });
+  }
+
+  setError() {
+    this.error = `${this.template.host.localName} Error`;
+  }
+
+  /**
+   * Whether the field value can be displayed.
+   * @returns {boolean}
+   */
+  get isValid() {
+    return this.validated && !this.error;
+  }
+
   renderedCallback() {
     // eslint-disable-next-line no-unused-expressions
-    this.hasTextToDisplay && this.setBadgeColors();
+    this.isValid && this.hasTextToDisplay && this.setBadgeColors();
   }
 
   /**
    * get the text content to display depending the provided props.
    * @returns {string}
    */
   get textToDisplay() {
```

Comment:
Could just remove this whole getter and declare hasTextToDisplay like this

```javascript
get hasTextToDisplay() {
  return !!(this.label || this.result?.raw?.[this.field]);
}
```

Reply:
sure 👌